src/mainloop.rkt

```diff
@@ -326,15 +329,20 @@
 (define (run-improve! initial specification context pcontext)
   (timeline-event! 'preprocess)
   (define-values (simplified preprocessing)
-    (find-preprocessing initial specification context))
+    (find-preprocessing initial specification context)) ;; Eventually we can remove the preprocessing return here, but that requires fixing 
+                                                        ;; calls here & the sandbox call & subsequent usage that still require it
```

Comment:
I would remove these comments from the PR, or move them to a shorter TODO comment in the place that needs actual fixing.